src/tests/cipher.cpp

```diff
@@ -696,7 +696,27 @@ TEST_F(rnp_tests, s2k_iteration_tuning)
     /// TODO test that hashing iters_xx data takes roughly requested time
 }
 
-bool
+TEST_F(rnp_tests, s2k_iteration_encode_decode)
+{
+    const size_t MAX_ITER = 0x3e00000; // 0x1F << (0xF + 6);
+    // encoding tests
+    assert_int_equal(pgp_s2k_encode_iterations(0), 0);
+    assert_int_equal(pgp_s2k_encode_iterations(512), 0);
+    assert_int_equal(pgp_s2k_encode_iterations(1024), 0);
+    assert_int_equal(pgp_s2k_encode_iterations(1024), 0);
+    assert_int_equal(pgp_s2k_encode_iterations(1025), 1);
+    assert_int_equal(pgp_s2k_encode_iterations(1088), 1);
+    assert_int_equal(pgp_s2k_encode_iterations(1089), 2);
+    assert_int_equal(pgp_s2k_encode_iterations(2048), 16);
+    assert_int_equal(pgp_s2k_encode_iterations(MAX_ITER), 0xFF);
+    // decoding tests
+    assert_int_equal(pgp_s2k_decode_iterations(0), 1024);
+    assert_int_equal(pgp_s2k_decode_iterations(1), 1088);
+    assert_int_equal(pgp_s2k_decode_iterations(16), 2048);
+    assert_int_equal(pgp_s2k_decode_iterations(0xFF), MAX_ITER);
```

Comment:
It would be good to add edge case(s) where value is larger then MAX_ITER in the encode call.